JM/CF/Scripts/5_Mission/CommunityFramework/Mission/MissionBase.c

```diff
@@ -1,5 +1,9 @@
 modded class MissionBase
 {
+#ifdef SERVER
+	float m_CF_UpdateTime;
+#endif
```

Comment:
Does `SERVER` define work for singleplayer e.g. COM? What if you disconnect afterward and connect to a server? I would suspect the code is preprocessed once on the game compilation, later role changes not affecting it.

Reply:
`SERVER` is only defined when running a dedicated server.

Reply:
So this solution does not work in single-player? If so that is not an option. We can't rely on that define then.

Reply:
Rate limiting is not supposed to be used on client or in SP, it only makes sense on a dedicated server. On client, the update loop may also be used for rendering, thus should go as fast as it can.